### ISSUE TITLE
autoIncrement doesn't work on non primary key fields

### ISSUE THREAD

**Issue by zeluis9** [balderdashy/sails-mysql#213]
autoIncrement feature doesn't work on non primary fields. The model has autoPK set to false and one integer column with autoIncrement. The column with autoIncrement set to true is not being incremented automatically and it's not being fulfilled at all. I currently have a workaround, but that seems very resource-expensive to me:
```javascript
beforeCreate: function(myModel, next) {
    if(!myModel.id) {
        model.count(function(err,num) {
            if(err){
                next(err);
            } else {
                myModel.id = num + 1;
                next();
            }
        });
    } else {
        next();
    }
}
```
Thanks in advance.

**Comment by devinivy** [comment]
Sounds a bit like a bug to me!  This should be taken care of around here in the codebase: https://github.com/balderdashy/sails-mysql/blob/1e7f0b4cf19d790e4f3bab0351335b77b3aeb6c2/lib/sql.js#L77

Will accept a PR.  Anyone embarking on a PR, bear in mind MySQL can only have one auto-incrementing column, and it must be indexed.  See http://stackoverflow.com/questions/14086324/mysql-innodb-autoincrement-non-primary-key

**Comment by zeluis9** [comment]
I think I can make an PR later (22h GMT+1)

**Comment by sailsbot** [comment]
Thanks for posting, @zeluis9.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
